From 73a4e2855c85c56c881d6a4c2dcf7887d9108095 Mon Sep 17 00:00:00 2001
From: Ryan QIAN <b32804@freescale.com>
Date: Tue, 22 Nov 2011 10:40:25 +0800
Subject: ENGR00162792-2 [IMX6Q] mmc host pm caps should be board related

- add board data for mmc host keep power at suspend

Signed-off-by: Ryan QIAN <b32804@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c    |    2 ++
 arch/arm/plat-mxc/include/mach/esdhc.h |    1 +
 2 files changed, 3 insertions(+)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index f1f9600..4274d3d 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -420,6 +420,7 @@ static const struct esdhc_platform_data mx6q_arm2_sd3_data __initconst = {
 	.wp_gpio = MX6Q_ARM2_SD3_WP,
 	.support_18v = 1,
 	.support_8bit = 1,
+	.keep_power_at_suspend = 1,
 	.delay_line = 0,
 	.platform_pad_change = plt_sd3_pad_change,
 };
@@ -428,6 +429,7 @@ static const struct esdhc_platform_data mx6q_arm2_sd3_data __initconst = {
 static const struct esdhc_platform_data mx6q_arm2_sd4_data __initconst = {
 	.always_present = 1,
 	.support_8bit = 1,
+	.keep_power_at_suspend = 1,
 	.platform_pad_change = plt_sd4_pad_change,
 };
 
diff --git a/arch/arm/plat-mxc/include/mach/esdhc.h b/arch/arm/plat-mxc/include/mach/esdhc.h
index 9ef53ab..1862c70 100644
--- a/arch/arm/plat-mxc/include/mach/esdhc.h
+++ b/arch/arm/plat-mxc/include/mach/esdhc.h
@@ -25,6 +25,7 @@ struct esdhc_platform_data {
 	unsigned int always_present;
 	unsigned int support_18v;
 	unsigned int support_8bit;
+	unsigned int keep_power_at_suspend;
 	unsigned int delay_line;
 	int (*platform_pad_change)(int clock);
 };
-- 
1.7.9.5

